src/main/java/edu/greenblitz/robotName/subsystems/LED/LED.java

```diff
@@ -0,0 +1,153 @@
+		this.addressableLED.start();
+		LEDBlinkTimer = new Timer();
+		noteOut = false;
+		noteInRobot = false;
+	}
+	
+	public static LED getInstance() {
+		init();
+		return instance;
+	}
+	
+	public static void init() {
+		if (instance == null) {
+			instance = new LED();
+		}
+	}
+	
+	public void startTimer() {
+		LEDBlinkTimer.restart();
+	}
+	
+	public double getTimerTime() {
+		return LEDBlinkTimer.get();
+	}
+	
+	
+	public void setLEDColor(Color color, Section section) {
+		setLEDColor(color, section.startIndex(), section.endIndex());
+	}
+	
+	public void setLEDColor(Color color, int startIndex, int endIndex) {
+		for (int i = startIndex; i < endIndex; i++) {
+			setLEDColor(color, i);
+		}
+	}
+	
+	public void setLEDColor(Color color, int index) {
+		this.addressableLEDBuffer.setLED(index, color);
+	}
+	
+	public void turnOff(int index) {
+		setLEDColor(new Color(0, 0, 0), index);
+	}
+	
+	public void turnOff(int startIndex, int endIndex) {
+		for (int i = startIndex; i < endIndex; i++) {
+			turnOff(i);
+		}
+	}
+	
+	public void turnOff(Section section) {
+		turnOff(section.startIndex(), section.endIndex());
+	}
+	
+	@Override
+	public void periodic() {
+		this.addressableLED.setData(addressableLEDBuffer);
+		blinkByNotePlace();
+		noteMode();
+		rumbleIfNoteOut();
+	}
+	
+	public void colorByMode() {
+		if (ScoringModeSelector.isAmpMode()) {
+			setLEDColor(LEDConstants.AMP_MODE_COLOR, LEDConstants.ALL_LED);
+		} else {
+			setLEDColor(LEDConstants.SHOOTER_MODE_COLOR, LEDConstants.ALL_LED);
+		}
+	}
+	
+	public void blinkIfInArm() {
+		if (LEDBlinkTimer.get() % (LEDConstants.BLINK_DURATION * 2) >= LEDConstants.BLINK_DURATION) {
+			turnOff(LEDConstants.ALL_LED);
+		} else {
+			setLEDColor(LEDConstants.AMP_MODE_COLOR, LEDConstants.ALL_LED);
+		}
+	}
+	
+	public void blinkIfInShooter() {
+		if (LEDBlinkTimer.get() % (LEDConstants.BLINK_DURATION * 2) >= LEDConstants.BLINK_DURATION) {
+			turnOff(LEDConstants.ALL_LED);
+		} else {
+			setLEDColor(LEDConstants.SHOOTER_MODE_COLOR, LEDConstants.ALL_LED);
+		}
+	}
+	
+	public void blinkByNotePlace() {
+		if (noteInRobot) {
+			if (Funnel.getInstance().isObjectIn()) {
+				new BlinkIfInShooter().schedule();
+			} else if (Roller.getInstance().isObjectInside()) {
+				new BlinkIfInArm().schedule();
+			}
+		}
+	}
+	
+	public void noteMode() {
+		if (Intake.getInstance().getExitBeamBreakerValue()
+				|| Intake.getInstance().getEntranceBeamBreakerValue()
+				|| Funnel.getInstance().isObjectIn()
+				|| Roller.getInstance().isObjectInside()) {
+			noteInRobot = true;
+		}
+		if (!(Intake.getInstance().getExitBeamBreakerValue())
+				&& !(Intake.getInstance().getEntranceBeamBreakerValue())
+				&& !(Funnel.getInstance().isObjectIn())
+				&& !(Roller.getInstance().isObjectInside())) {
+			noteOut = true;
+		}
+		else{
+			noteOut = false;
+		}
+	}
+	
+	public void rumbleIfNoteOut() {
+		if (noteInRobot) {
+			if (noteOut) {
+				new Rumble().schedule();
+				noteInRobot = false;
+			}
+		}
+	}
```

Comment:
yoav  - look again at these functions, maybe think of another way to do it like a default command